packages/snaps-utils/src/snaps.ts

```diff
@@ -321,3 +307,25 @@ export function isCaipChainId(chainId: unknown): chainId is string {
     )
   );
 }
+
+/**
+ * Utility function to check if an origin has permission (and caveat) for a particular snap.
+ *
+ * @param permissions - An origin's permissions object.
+ * @param snapId - The id of the snap.
+ * @returns A boolean based on if an origin has the specified snap.
+ */
+export function hasSnap(
```

Comment:
This function name feels a bit strange. Perhaps something like `isSnapPermitted` would be better?